File: model/src/main/java/io/github/runtimemodels/chazm/entity/CapabilityEntity.java
package io.github.runtimemodels.chazm.entity;

import com.google.inject.assistedinject.Assisted;
import io.github.runtimemodels.chazm.id.UniqueId;

import javax.inject.Inject;
import javax.validation.constraints.NotNull;

class CapabilityEntity extends AbstractEntity<Capability> implements Capability {

    @Inject
    CapabilityEntity(@NotNull @Assisted final UniqueId<Capability> id) {
        super(id);
    }

    @Override
    public boolean equals(final Object object) {
        if (object instanceof Capability) {
            final Capability capability = (Capability) object;
            return super.equals(capability);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return super.hashCode();
    }

    @Override
    public String toString() {
        return super.toString();
    }

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/entity/CapabilityEntity.kt
package runtimemodels.chazm.model.entity

import com.google.inject.assistedinject.Assisted
import runtimemodels.chazm.api.entity.Capability
import runtimemodels.chazm.api.id.CapabilityId
import javax.inject.Inject

internal data class CapabilityEntity @Inject constructor(
    @Assisted override val id: CapabilityId
) : Capability
